### DISCUSSION TITLE
correct way to release a connection

### DISCUSSION THREAD

**Discussion by naveen1337** [sidorares/node-mysql2#1536] [Q&A]
In my code I release a connection on finally in try catch block, but if there are any error occurred in while get  a connection, the finally section gives an error because the release function will not be there, I can fix that by checking is the connection variable it not null before calling release function. Is there any good way to do that ? 

```js
    // is that work ? 
      connection.release?.();
```

```js 
async create(payload: any) {
    let connection = null;
    try {
      const sqlQuery = qb(this.table).insert(payload).toString();
      connection = await dbConnection.getConnection();
      const response = await connection.execute(sqlQuery);
      if (response) {
        return successResponse(response[0]);
      } else {
        failResponse(`failed to create ${this.name}`);
      }
    } catch (err) {
      console.log(err);
      return {
        status: false,
      };
    } finally {
      connection.release();
    }
  }
```

**Comment by sidorares** [comment]
`if (connection) { connection.release() }` should probably work

Is there any reason why `dbConnection.execute()` is not good for you? Assuming that `dbConnection` is `PromisePool` instance
